### PR TITLE
Add mitX Online as an alternative social auth backend

### LINKED ISSUE #4989: mitX Online Integration: Add social auth backend
Add a social auth backend for mitX Online, this should largely be the same as the edx.org one, with some differences:

### Design
- If the learner has an mitX Online social auth, always use that as the source of truth (i.e. the logic to update the email address from edx.org should only be run if that’s the only social auth for that learner).
- The mitX Online social auth should pull user profile data from the mitX Online user api not the OpenedX user api. If there’s gaps here, let’s discuss, but I believe MicroMasters should be ok as long as it’s getting at least basic email/name data.
- For now, the edx.org social auth should continue to be the default one.

### Acceptance Criteria
- [x] The learner should be able to create a new account with edx.org or mitX Online
- [x] The learner should be able to login with existing accounts for all combinations of having edx.org and/or mitX Online social auths.
- [x] The learner should be able to register with both edx.org and mitX Online using an existing MicroMasters account. Social auth should provide this behavior out-of-box, but we should verify this is configured and functions without issue.

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Closes #4989 

#### What's this PR do?
This PR adds mitx online as an alternative social auth backend. It does not expose this in the UI yet.

#### How should this be manually tested?
Follow the [setup steps](https://github.com/mitodl/micromasters/pull/4996/files#diff-d64c15ff6ea885e0d24eed19dd5c1db82759fffa0f15c43f029ddd8853882fba)

Verify the following authentication flows work:

- Can register a new "edx.org" account
- Can register a new mitx online account
- Can add mitx online account to an existing "edx.org" account
- For the following account types, verify you can still login:
  - an account with only an "edx.org" social auth
  - an account with only an mitxonline social auth
  - an account with both kinds of social auth (verify that you can sign in with both platforms)

### CODE REVIEW

File: docs/configure_open_edx.md [hunk 1]
```diff
@@ -132,6 +137,19 @@ $ make requiements
...
+  - In Open edX:
+    - Go to `/admin/oauth2_provider/application/` and verify that an application named `micromasters-mitxonline` (name does not really matter here, but must be different from the one in the section above) exists with these settings:
+      - `Redirect uris`: `http://mm.odl.local:8079/complete/mitxonline/`
+      - `Client type`: "Confidential"
+      - `Authorization grant type`: "Authorization code"
+      - `Skip authorization`: checked
+      - Other values are arbitrary but be sure to fill them all out. Save the client id and secret for later
+  - In Micromasters:
+    - Set `MITXONLINE_BASE_URL` to the correct URL that is accessible from Micromasters container and host, e.g. `http://edx2.odl.local:18000/`
+    - Set `MITXONLINE_API_CLIENT_ID` to the client id
```
Comment (annagav) [inline docs/configure_open_edx.md:154]: This should be `MITXONLINE_CLIENT_ID` to match the settings

**Comment by annagav** [comment]
@rhysyngsun the fix worked for me. Thank you!

Another question: When a new learner comes with the intention of signing up for DEDP, are they going have a choice which backend to signup with? Do they have to create an account with edxorg if they do not have it already?

**Comment by rhysyngsun** [comment]
> Another question: When a new learner comes with the intention of signing up for DEDP, are they going have a choice which backend to signup with? Do they have to create an account with edxorg if they do not have it already?

Yes, we'll be adding this, but in another PR.

**Comment by rhysyngsun** [comment]
@annagav this should be good to review now

**Comment by annagav** [comment]
@rhysyngsun when I am registering with mitX online do I have to enter same username and email as I am registered on edxorg?

**Review by annagav — APPROVED** [review]
Everything works for me 👍

**Pull request merged**